src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,104 @@
-export const Pagination = () => {};
+import React from 'react';
+import cn from 'classnames';
+
+type Props = {
+  items: string[],
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  items,
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pagesCount = Math.ceil(total / perPage);
+  const pages = [];
+  const isFirstPage = currentPage === 1;
+  const isLastPage = currentPage === pagesCount;
+
+  for (let i = 1; i <= pagesCount; i += 1) {
+    pages.push(i);
+  }
+
+  const prevPage = (page: number) => () => {
+    if (page > 0 && page !== 1) {
+      onPageChange(page - 1);
+    }
+  };
+
+  const nextPage = (page: number) => () => {
+    if (page < pagesCount && page !== pagesCount) {
+      onPageChange(page + 1);
+    }
+  };
+
+  return (
+    <>
+      <ul className="pagination">
+        <li
+          className={cn(
+            'page-item',
+            { disabled: isFirstPage },
+          )}
+        >
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={isFirstPage}
+            onClick={prevPage(currentPage)}
+          >
+            «
+          </a>
+        </li>
+        {pages.map(page => (
+          <li
+            key={page}
+            className={cn(
+              'page-item',
+              { active: page === currentPage },
+            )}
+          >
+            <a
+              data-cy="pageLink"
+              className="page-link"
+              href="#1"
```

Comment:
The href attribute should be dynamic based on the page number, not a static '#1'.